core/templates/domain/topic/subtopic.model.ts

```diff
@@ -13,22 +13,23 @@
 // limitations under the License.
 
 /**
- * @fileoverview Factory for creating and mutating instances of frontend
+ * @fileoverview Model class for creating and mutating instances of frontend
  * subtopic domain objects.
  */
 
-import { ShortSkillSummary } from
-  'domain/skill/short-skill-summary.model';
-
-import constants from 'assets/constants';
+import { ShortSkillSummary } from 'domain/skill/short-skill-summary.model';
+import { AppConstants } from 'app.constants';
 
 export interface SubtopicBackendDict {
   'id': number;
   'title': string;
   'skill_ids': string[];
-  'thumbnail_filename': string;
-  'thumbnail_bg_color': string;
-  'url_fragment': string;
+  // For a newly created subtopic, properties below are initialized
+  // with a null value and should be passed by user to create a subtopic.
+  // Here null implies that properties are yet to be assigned.
+  'thumbnail_filename': string | null;
```

Comment:
If you are doing this correctly, you will not need null anywhere here, since all subtopic objects (either created in the frontend or passed from the backend) will always have every field fully populated.